### PR TITLE
simplify lookup for non-param substring

### PR DESCRIPTION
**Pull request by koenpunt**
In addition to #100, an suggestion with a simplified lookup for non-param substring

### CODE REVIEW

**Comment by koenpunt** [comment]
@dannyvankooten can you take a look at this?

**Comment by dannyvankooten** [comment]
This looks great @koenpunt. I'm not really running AltoRouter in production so haven't tested this on large sub-sets of routes but from what I can tell, this should just work. Tests agree, and we have one for every route format AFAIK.

Running this through the same benchmark described in #130 shows a nice performance improvement as well (for routers with many static routes).

```
Time: 0.42830586433411 seconds
Peak memory usage: 2MB
```

👍 for merging this.

File: tests/AltoRouterTest.php [hunk 1]
```diff
@@ -292,6 +292,33 @@ public function testMatchWithFixedParamValues()
...
 	}
 	
+	public function testMatchWithPlainRoute()
+	{
+		$router = $this->getMockBuilder('AltoRouterDebug')
+			->setMethods(array('compileRoute'))
+			->getMock();
+		
+		// this should prove that compileRoute is not called when the route doesn't
+		// have any params in it, but this doesn't work because compileRoute is private.
```
Comment (koenpunt) [inline tests/AltoRouterTest.php]: @dannyvankooten have any suggestions on how to test this? Should we make the method protected?

Comment (dannyvankooten) [inline tests/AltoRouterTest.php]: @koenpunt I'm not a big fan of testing private methods, we should test the public API of the class and leave the rest up to the internals of the class. As this is mainly a performance thing and expected output remains the same whether `compileRoute` is called or not, I think we shouldn't force to get that covered by tests.

I prefer an untested private method than making the method `protected` or even `public` and then having to support it for years.

**Pull request merged**